### ISSUE TITLE
[MLForecast] Calling preprocess on test set changes the future

### ISSUE THREAD

**Issue by JLiekenbrock** [Nixtla/mlforecast#501]
### What happened + What you expected to happen

Calling preprocess on a test set changes the result of the future.
I would expect preprocess to be side effect free.

### Versions / Dependencies

MacOS, Python3.12.3, MLForecast:1.0.2

### Reproduction script

from mlforecast.utils import generate_series, generate_prices_for_series
from mlforecast import MLForecast
import lightgbm as lgb
import pandas as pd

series = generate_series(200, equal_ends=True)

def train_test_split_last_n_pandas(df: pd.DataFrame, time_col: str = "ds", id_col: str = "unique_id", n: int = 3):
    df_sorted = df.sort_values(by=[id_col, time_col])
    test = df_sorted
    train = df_sorted.groupby(id_col).apply(lambda group: group.iloc[:-n] if len(group) > n else group.iloc[0:0])
    train = train.reset_index(drop=True)
    return train, test

train, test = train_test_split_last_n_pandas(series, n =3)

fcst = MLForecast(
    models=[lgb.LGBMRegressor()],
    freq='D',
    lags=[1],
)

fcst.fit(train)

fu1 = fcst.make_future_dataframe(h=1)

pre = fcst.preprocess(test)

fu2 = fcst.make_future_dataframe(h=1)

print(fu1.equals(fu2))

### Issue Severity

Low: It annoys or frustrates me.

**Comment by JLiekenbrock** [comment]
Okay, it seems that this is on purpose and allows updating the past data known to the model to allow moving forward in time, is that correct?

Simple example here, where i move the prediction starting point forwards after training:

`
from datetime import timedelta
import polars as pl

offset = 3
horizon = 28-offset
forecast_start = valid_start_date + timedelta(days=offset)
forecast_end = forecast_start + timedelta(days=horizon - offset)

extended_long = long.filter(pl.col("date") <= forecast_start)

_ = fcst.preprocess(
    extended_long.drop(["d", "wm_yr_wk"]),
    id_col="id",
    time_col="date",
    target_col="y",
    static_features=categoricals,
    return_X_y=True,    
    as_numpy=True,
)

X_df = fcst.make_future_dataframe(h=horizon)
X_df = X_df.filter(pl.col("date") >= forecast_start)

print(min(fcst.make_future_dataframe(h=horizon)['date']))

static = (
    long
    .filter(pl.col("date") >= valid_start_date)
    .drop(['item_id', 'dept_id', 'cat_id', 'store_id', 'state_id'] + ["y", "d", "wm_yr_wk"])
)
X_df = X_df.join(static, on=["id", "date"], how="left").drop()

print(fcst.get_missing_future(horizon, X_df))
new_df = (
    long.drop(["d", "wm_yr_wk"])
        .filter((pl.col("date") >= forecast_start) & (pl.col("date") <= forecast_end))
)

fcst.models_ = {"LGBMRegressor": model}  # Only if not already set
preds = fcst.predict(h=horizon, X_df=X_df)
`
